### PR TITLE
[fix](Nereids): don't pushdown project when project contains both side of join

### PR DESCRIPTION
**Pull request by keanji-x**
## Proposed changes

if project contains both side of join, it can't be pushed.
such as:
Project(l, null as r)
------ L(l) left anti join R(r)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run p0

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/join/PushDownProjectThroughSemiJoin.java [hunk 1]
```diff
@@ -72,12 +75,27 @@ public List<Rule> buildRules() {
...
+                        }
                         Plan newRight = pushdownProject(project);
                         return topJoin.withChildren(topJoin.left(), newRight);
                     }).toRule(RuleType.PUSH_DOWN_PROJECT_THROUGH_SEMI_JOIN_RIGHT)
                 );
     }
 
+    private boolean projectBothJoinSide(LogicalProject<LogicalJoin<GroupPlan, GroupPlan>> project) {
+        // if project contains both side of join, it can't be pushed.
+        // such as:
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/join/PushDownProjectThroughSemiJoin.java:89]: select

**Comment by XieJiann** [comment]
run p0

**Comment by XieJiann** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by jackwener — APPROVED** [review]

**Pull request merged**